lowercase code in basics/memory

wilzbach opened the pull request.

wilzbach commented on basics/memory.md:
```diff
@@ -1,8 +1,8 @@
-manage. However manual memory management is very error-prone and thus
-D uses a *garbage collector* per default to free unused memory.
+D is a system programming language and thus allows manual
+memory management. However manual memory management is very error-prone
```
I am not sure about this change as it repeats manual memory management, but the state before is definitely wrong.

JackStouffer commented on basics/memory.md:
```diff
@@ -1,8 +1,8 @@
-D uses a *garbage collector* per default to free unused memory.
+D is a system programming language and thus allows manual
+memory management. However manual memory management is very error-prone
+and thus D uses a *garbage collector* per default to free unused memory.
```
This is misleading. The GC only collects memory allocated with the GC.

wilzbach commented:
> This is misleading. The GC only collects memory allocated with the GC.

Good catch (in the existing content). Changed the content accordingly and rebased.
